src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/naxxramas/boss_four_horsemen.cpp

```diff
@@ -100,6 +100,10 @@ struct boss_horsmenAI : public ScriptedAI
     boss_horsmenAI(Creature* creature) : ScriptedAI(creature)
     {
         m_instance = (ScriptedInstance*)creature->GetInstanceData();
+        m_creature->GetCombatManager().SetLeashingCheck([&](Unit*, float x, float y, float /*z*/)
+        {
+            return ( x > 2577.3 && y < -3024.0f ) || ( x > 2585.9f && y < -3015.1f ) || ( x > 2594.6f && y < -3006.7f );
```

Comment:
Looking from the outside of the boss arena towards the bosses the first () is the left corner of the door, the second () is the center of the door and the third () is the right corner of the door.

Reply:
Consider to comment this in code, so its easier to understand when the next guy does not have this PR at hand <3